source/function.cpp

```diff
@@ -86,8 +86,8 @@ pair<string, string> function_arguments_for_lambda(clang::FunctionDecl const *re
 		QualType qt = record->getParamDecl(i)->getOriginalType().getCanonicalType();
 		r += qt.getAsString() + ' ';
 		if( !qt->isReferenceType()  and  !qt->isPointerType() ) r += !qt.isConstQualified() ? "const & " : "& ";
-		r += "a" + std::to_string(i);
-		a += "a" + std::to_string(i);
+		r += string( record->getParamDecl(i)->getName() );
+		a += string( record->getParamDecl(i)->getName() );
```

Comment:
Have you checked if this approach works when there is no name of argument? Like `MyClass(int, float)`

Reply:
Nope. I forgot that. But why aren't you worried about the same thing but for the complete constructor with all arguments?

Reply:
Anyway I reverted this change. I do not need it to get the argument name in the documentation in python environment.

Reply:
hm... i think there was some miscommunication here: i was not saying that this change is bad (i think it good!) - i was just asking if this change would break for cases when no name of arg was giving. Any chance you can re-introduce this change back? And if you can not check case of args-without-name - no worries, i will just check that locally. Thanks,

Reply:
Thank you.